wire/msgtx.go

```diff
@@ -326,6 +413,13 @@ func (msg *MsgTx) Deserialize(r io.Reader) error {
 	return msg.BtcDecode(r, 0)
 }
 
+// DeserializeWitness decodes a transaction from r into the receiver, where the
+// encoded transaction format within r may possibly utilize the new serialization
+// format created to encode transactions bearing witness data within inputs.
+func (msg *MsgTx) DeserializeWitness(r io.Reader) error {
+	return msg.BtcDecode(r, WitnessVersion)
```

Comment:
As mentioned above the transaction version should be used instead and then there is no need for a separate function.  Plus, using the `WitnessVersion` is incorrect since `DeserializeX`, as mentioned elsewhere, is intended for the long term stable storage format for the database to which the _network_ protocol version must not apply.  It's fine (and preferable) if they happen to be the same encoding, but the distinction must be kept because future versions can change the serialization format on the wire, but it still must be able to read whatever the database format is.